tasks/ssl.yml

```diff
@@ -0,0 +1,22 @@
+- block:
+    - name: create ssl key
+      openssl_privatekey:
+        path: "{{ openstack_undercloud_ssl_key }}"
+        size: "{{ openstack_undercloud_ssl_key_size }}"
+        type: "{{ openstack_undercloud_ssl_key_type }}"
+      register: _generate_ssl_key
+
+    - name: check for certificate
+      stat:
+        path: "{{ openstack_undercloud_ssl_certificate }}"
+      register: _ssl_certificate
+
+    - name: create SSL certificiate
+      command: >-
```

Comment:
Can the `openssl_certificate` module be used here?

Reply:
I could not find a way to directly create a self-signed cert in one step. `openssl_certificate` required creating first a CSR, and the docs I was handed did not take that step. I'd be happy to consider updating this in the future if someone else knows more about SSL certificates than I do and can guide me in that.

Reply:
The whole process *should* be able to be achieved using modules, e.g.:

```
- hosts: localhost
  vars_prompt:
    - name: common_name
      prompt: common name
      private: false
  vars:
    gen_pubkey: false
    pubkey: "./ssl-certs/{{ common_name }}.pub"
    privkey: "./ssl-certs/{{ common_name }}.key"
    csr: "/tmp/{{ common_name }}.csr"
    cert: "./ssl-certs/{{ common_name }}.crt"
  tasks:
    - name: generate private key
      openssl_privatekey:
        path: "{{ privkey }}"
        size: 4096
        force: true

    - name: generate public key from private key
      openssl_publickey:
        path: "{{ pubkey }}"
        privatekey_path: "{{ privkey }}"
        force: true
      when: gen_pubkey

    - name: generate csr
      openssl_csr:
        path: "{{ csr }}"
        privatekey_path: "{{ privkey }}"
        common_name: "{{ common_name }}"
        force: true

    - name: Generate a Self Signed OpenSSL certificate
      openssl_certificate:
        path: "{{ cert }}"
        privatekey_path: "{{ privkey }}"
        csr_path: "{{ csr }}"
        provider: selfsigned
        force: true
```

Reply:
This results in an intermediate CSR file that there is no need for. The modules have no way to directly generate a self-signed certificate without an existing CSR file. OpenStack docs specify skipping the CSR (or, at least, they ignore it and take the single-step generation process).

Reply:
I do not think that creating a temporary CSR file (which is a very normal thing to do when generating certificates) is a big deal, in spite of what the OpenStack docs say.  Creating a temp file is less bad than avoiding Ansible built-ins IMHO.